core/src/appleMain/kotlin/Peripheral.kt

```diff
@@ -82,7 +82,10 @@ import kotlin.coroutines.cancellation.CancellationException
 public actual fun CoroutineScope.peripheral(
     advertisement: Advertisement,
     builderAction: PeripheralBuilderAction,
-): Peripheral = peripheral(advertisement.cbPeripheral, builderAction)
+): Peripheral {
+    advertisement as CBPeripheralCoreBluetoothAdvertisement
```

Comment:
Same here. We really shouldn't allow the function to accept _any_ `Advertisement` if it _can't_.